pkg/minishift/provisioner/minishift_provisioner.go

```diff
@@ -91,6 +91,12 @@ func (provisioner *MinishiftProvisioner) Package(name string, action pkgaction.P
 func (provisioner *MinishiftProvisioner) dockerDaemonResponding() bool {
 	log.Debug("checking docker daemon")
 
+	// Start the docker daemon service if not running.
+	if out, err := provisioner.SSHCommand("sudo systemctl -f start docker"); err != nil {
+		log.Warnf("Error getting SSH command to check if the daemon is running: %s", err)
+		log.Debugf("'sudo systemctl docker start' output:\n%s", out)
```

Comment:
@praveenkumar Why we are not failing here. Because if we can not start docker then anyway it is going to fail.

Reply:
the `docker version` would fail anyway afterwards

Reply:
@gbraad @LalatenduMohanty yes I did it because it will anyway fail in the later stage but yes I can fail it early.

Reply:
f you look at the whole block of code, maybe returning the `err` would be a better idea anyway. as in that case the `logf` and `warnf` can be isolated to a single occurence in the caller (there), instead of the called function (here)

Reply:
@gbraad return err from here is not a good idea because then we are not going to use `mcnutils.WaitFor(provisioner.dockerDaemonResponding)` functionality which actually tries to get proper info after 30 attempts. I do agree can return `false` here so that it will not give user false impression of failing for version but instead it's failing due to service not running.